app/policies/group_policy.rb

```diff
@@ -16,7 +17,7 @@ def resolve
       if user.super_admin?
         scope.all
       else
-        scope.where(organisation_id: user.organisation_id)
+        scope.for_user(user)
```

Comment:
Super_admins can access every group.

Everyone else can only access the groups they belong to.